From 72716ee4957999c0a7c66eae2b11e4e7fee30b92 Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Mon, 8 Sep 2014 13:05:48 -0400
Subject: [tools] perf/timechart: Fix off-by-one error in 'record' argv handling

Message-id: <1410181752-25631-104-git-send-email-jolsa@redhat.com>
Patchwork-id: 91858
O-Subject: [PATCH RHEL7.1 BZ1134356 103/307] perf timechart: Fix off-by-one error in 'record' argv handling
Bugzilla: 1134356
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Don Zickus <dzickus@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1134356
https://bugzilla.redhat.com/show_bug.cgi?id=1134356

upstream
========
commit 263f89bf7d0f5ba98077dda8df1ff814862ad5ba
Author: Ramkumar Ramachandra <artagnon@gmail.com>
Date: Sun Mar 16 12:06:05 2014 -0400

description
===========
Since 367b315 (perf timechart: Add support for -P and -T in timechart
recording, 2013-11-01), the 'perf timechart record' command stopped
working:

  $ perf timechart record -- git status
  Workload failed: No such file or directory

This happens because of an off-by-one error while preparing the argv for
cmd_record(): it attempts to execute the command 'status' and complains
that it doesn't exist. Fix this error.
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/perf/builtin-timechart.c b/tools/perf/builtin-timechart.c
index 25526d6..d4991a2 100644
--- a/tools/perf/builtin-timechart.c
+++ b/tools/perf/builtin-timechart.c
@@ -1238,7 +1238,7 @@ static int timechart__record(struct timechart *tchart, int argc, const char **ar
  for (i = 0; i < old_power_args_nr; i++)
   *p++ = strdup(old_power_args[i]);
 
- for (j = 1; j < (unsigned int)argc; j++)
+ for (j = 0; j < (unsigned int)argc; j++)
   *p++ = argv[j];
 
  return cmd_record(rec_argc, rec_argv, NULL);
-- 
1.7.1